sky/cli.py

```diff
@@ -3113,25 +3121,39 @@ def _output():
                 spot_price_str = f'$ {item.spot_price:.3f}' if not pd.isna(
                     item.spot_price) else '-'
                 region_str = item.region if not pd.isna(item.region) else '-'
-                accelerator_table_vals = [
-                    item.accelerator_name,
-                    item.accelerator_count,
-                    item.cloud,
-                    instance_type_str,
-                    device_memory_str,
-                    cpu_str,
-                    host_memory_str,
-                    price_str,
-                    spot_price_str,
-                ]
-                if not show_all:
-                    accelerator_table_vals.append(region_str)
-                accelerator_table.add_row(accelerator_table_vals)
-
-            if i != 0:
+
+                requested_accelerator_count = (item.accelerator_count
+                                               if not gpu_name_and_count else
+                                               int(split_gpu_name[1]))
+
+                if requested_accelerator_count == item.accelerator_count:
+                    accelerator_table_vals = [
+                        item.accelerator_name,
+                        item.accelerator_count,
+                        item.cloud,
+                        instance_type_str,
+                        device_memory_str,
+                        cpu_str,
+                        host_memory_str,
+                        price_str,
+                        spot_price_str,
+                    ]
+                    if not show_all:
+                        accelerator_table_vals.append(region_str)
+                    accelerator_table.add_row(accelerator_table_vals)
```

Comment:
I feel this filtering should happen in service_catalog.list_accelerators. We can add a `quantity: Optional[int]` argument to `service_catalog.list_accelerators` and filter the rows returned if `quantity` is provided. By design, `cli.py` should have minimal logic.